rfc-126-custom-configuration-for-dependabot.md

```diff
@@ -0,0 +1,246 @@
+      - match: { dependency_name: markdown-toolbar-element }
+      - match: { dependency_name: miller-columns-element }
+      - match: { dependency_name: paste-html-to-govspeak }
+```
+
+### Keeping the configuration up to date
+
+Most of our applications use the same internal and framework libraries, so we'll want a single global configuration across all our apps. At the same time, there might be some libraries used by particular apps which fit within one of the three categories.
+
+Maintaining this file across each of our apps has the potential for it to become out of date with our global configuration. To solve this problem, this RFC proposes that we add functionality to [govuk-saas-config] which ensures the configuration file in each repo is up to date with a single global configuration file. Keeping up to date means ensuring that the individual repo configuration file contains at least the global configuration options plus any extra custom configuration for the app itself.
+
+Due to the fact that the configuration file is stored in the Git repo, [govuk-saas-config] will need to raise a PR to keep the configuration in sync rather than writing to the repo directly. This is similar to a script we have for [upgrading Ruby].
+
+[govuk-saas-config]: https://github.com/alphagov/govuk-saas-config
+[upgrading Ruby]: https://github.com/thomasleese/upgrade-ruby-version/blob/7d43077a10c732fc285e0bd6daad2d2c5740e352/main.py#L203-L207
+
+### Possible alternatives
+
+#### Auto-merge
+
+The proposal in this RFC [came from an earlier discussion of enabling auto-merge on our PRs][auto-merge-discussion]. Auto-merge is a feature which means [version bump PRs will automatically be merged][auto-merge] by Dependabot provided the tests are passing and the necessary approvals have been received.
+
+The idea is that by auto-merging our PRs, we wouldn't have to spend so much time dealing with them. However, there are a few reasons why this isn't possible at the moment:
+
+- We require approvals for all our PRs. Merging in the PR isn't what takes up the time, so having them merge automatically would only shave off a small amount of time.
+- The auto-merge functionality has to be enabled across the entire Dependabot account. We use a shared account across GDS. For this reason, and others, we [weren't able to turn it on][auto-merge-gds-way].
+
+[auto-merge]: https://dependabot.com/blog/automatic-pull-request-merging/
+[auto-merge-discussion]: https://docs.google.com/document/d/18P86EFSZdG8Uv5scbGzVoO8KhA3hkKp80UlzNoIctNE/edit
+[auto-merge-gds-way]: https://github.com/alphagov/gds-way/pull/428#discussion_r426502446
+
+#### Limit updates to lockfile
+
+Another option might be to limit Dependabot to only raise PRs against the `Gemfile.lock` file. The lockfile is used to pin a specific version, whereas the `Gemfile` can contain a wider version range (for example `~> 1.0` means `1.0 all the way up to but not including 2.0`).
```

Comment:
We could add a little more to this approach: gems we want to update would have little or no version restriction; gems we don't want to update would have a (more) precise restriction, to prevent Dependabot PRs, and signify that we're content with basically any working version of the gem.

Reply:
Thanks! Added in d984dab.